ethereum/contracts/Splitter.sol

```diff
@@ -1,28 +1,43 @@
-pragma solidity >= 0.5.7;
-pragma experimental ABIEncoderV2;
+pragma solidity >=0.5.8;
 
 contract Splitter {
-  mapping(address => bool) accounts;
+  mapping(address => bool) blackAccounts;
   address owner;
 
   constructor() public {
     owner = msg.sender;
   }
 
-  event Transfer(address from, bool status, uint balance1, uint balance2);
+  event Transfer(address from, address receiver1, address receiver2);
 
-  function sendEther(address[] memory receivers) public payable {
+  function sendEther(address payable receiver1, address payable receiver2) public payable {
     // One transfer call needs 2300 gas
     require(msg.sender.balance + 4600 > msg.value, 'A sender should have enough balance');
-    require(receivers.length == 2, 'The number of receivers should be 2');
-    require(msg.value % 2 == 0, 'The value should be evenly divisible');
     require(msg.value > 0, 'A given value should be bigger than 0');
-    require(msg.sender != receivers[0] && msg.sender != receivers[1], 'A sender should not be one of receivers');
+    require(msg.sender != receiver1 && msg.sender != receiver2, 'A sender should not be one of receivers');
+
+    if (blackAccounts[receiver1] || blackAccounts[receiver2]) {
+      revert('A receiver is in black list');
+    }
 
     uint value = msg.value / 2;
-    address(uint160(receivers[0])).transfer(value);
-    address(uint160(receivers[1])).transfer(value);
 
-    emit Transfer(msg.sender, true, receivers[0].balance, receivers[1].balance);
+    // TODO: security/no-send: Consider using 'transfer' in place of 'send'.
+    if (!receiver1.send(value)) {
+      blackAccounts[receiver1] = true;
+      revert('Transaction with receiver 1 has been failed');
+    }
+
+    if (!receiver2.send(value)) {
+      blackAccounts[receiver2] = true;
+      revert('Transaction with receiver 2 has been failed');
```

Comment:
Did you consider that receiver1 does not get paid if this decides to revert because receiver2 misbehaved?

I don't recall if I mentioned previously that the receivers may be contracts, and the default for a contracts is to reject `send` and `transfer`, so this is not far-fetched. We have to address it.